pkg/engine/vulnerability_builder_test.go

```diff
@@ -526,3 +606,171 @@ func TestGetBracketValues(t *testing.T) {
 		})
 	}
 }
+
+// TestGetAdjacents tests the functions [GetAdjacents()] and all the methods called by them
+func TestGetAdjacents(t *testing.T) { //nolint
+	type args struct {
+		idx   int
+		adj   int
+		lines []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want model.VulnLines
+	}{
+		{
+			name: "test_start_of_file",
+			args: args{
+				idx: 0,
+				adj: 3,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{1, 2, 3},
+				Lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+				},
+			},
+		},
+		{
+			name: "test_end_of_file",
+			args: args{
+				idx: 3,
+				adj: 3,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{3, 4, 5},
+				Lines: []string{
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+		},
+		{
+			name: "test_midle_of_file",
+			args: args{
+				idx: 1,
+				adj: 3,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{1, 2, 3},
+				Lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+				},
+			},
+		},
+		{
+			name: "test_even_adj",
+			args: args{
+				idx: 1,
+				adj: 2,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{2, 3},
+				Lines: []string{
+					"secondline",
+					"thirdline",
+				},
+			},
+		},
+		{
+			name: "test_even_adj_first_line",
+			args: args{
+				idx: 0,
+				adj: 2,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{1, 2},
+				Lines: []string{
+					"firstline",
+					"secondline",
+				},
+			},
+		},
+		{
+			name: "test_one_adj",
+			args: args{
+				idx: 3,
+				adj: 1,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{4},
+				Lines: []string{
+					"forthline",
+				},
+			},
+		},
+		{
+			name: "test_adj_bigger_than_file",
+			args: args{
+				idx: 3,
+				adj: 5,
+				lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+			want: model.VulnLines{
+				Positions: []int{1, 2, 3, 4},
+				Lines: []string{
+					"firstline",
+					"secondline",
+					"thirdline",
+					"forthline",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := getAdjacents(tt.args.idx, tt.args.adj, tt.args.lines)
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getAdjacents() = %v, want = %v", got, tt.want)
```

Comment:
use `test.StringifyStruct` to make the output easier to inspect in case of failure

Reply:
resolved